### PR TITLE
Update associations

### PR DESCRIPTION
**Pull request by emmanuelsteven**
In this PR 
- Created a Classroom class
- Created the has-many/belongs-to relationship between Classroom and Student. The following was implemented:
- Created the has-many side (a classroom has many students).
- Created the belongs-to side (a student belongs to a classroom).
- Created a Book class 
- Created Rental class 
- Created many-to-many relationship between Person and Book using the intermediate class Rental. The following was implemented:
- Created the has-many side of Book and Rental
- Created the belongs-to side of Rental and Book
- Created the has-many side of Person and Rental
- Created the belongs-to side of Rental and Person`
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @emmanuelsteven 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._
- [ ] [REQUIRED] I am adding this recommendation here because it is not in the file changes. In the Student class, when setting the classroom for a student it also adds it to the classrooms' students. There needs to be a method that serves as the 'belongs-to' side (a student belongs to a classroom). Take a look at this [example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

- [ ] [REQUIRED] Also,  I am adding this recommendation here because it is not in the file changes. There is also a part of the project requirement that says: Create the has-many side of Person and Rental (a person has many rentals). In the person class, @rentals should be initialized, also a method for adding rentals should be present. Take a look at this [example](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`). Kindly resolve the issues before requesting for another review.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 1]
```diff
@@ -20,6 +21,10 @@ def correct_name
     @name
   end
 
+  def add_rental(rental)
```
Comment (Shedrack-Sunday) [inline person.rb]: - [ ] Rental is a class. In adding a new rental, you have to add the parameters that initialize it.

### Sample code 


 Check this [guide](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)
 ``` def add_rental(book, date)
    # The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @emmanuelsteven        👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the ruby syntax✔️ 
- Kudus to heeding to the last reviewers comment✔️   
  
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉✔🍾**
Hello @emmanuelsteven 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Descriptive PR.
- Good commit messages.
- Correct git-flow.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**